src/sweeps/bayes_search.py

```diff
@@ -339,30 +386,23 @@ def _construct_gp_data(
     current_X: ArrayLike = []
     y: ArrayLike = []
 
-    # we calc the max metric to put as the metric for failed runs
-    # so that our bayesian search stays away from them
-    worst_metric: floating = np.inf if goal == "maximize" else -np.inf
-    for run in runs:
-        if run.state == RunState.finished:
-            try:
-                run_extremum = run.metric_extremum(
-                    metric_name, kind="minimum" if goal == "maximize" else "maximum"
-                )
-            except ValueError:
-                run_extremum = 0.0  # default
-            worst_metric = worst_func(worst_metric, run_extremum)
-    if not np.isfinite(worst_metric):
-        worst_metric = 0.0
-
     X_norms = params.convert_runs_to_normalized_vector(runs)
+    worst_metric = impute(goal, metric_name, "worst", runs=runs)
     for run, X_norm in zip(runs, X_norms):
-        if run.state == RunState.finished:
+        if run.state in [
+            RunState.finished,
+            RunState.killed,
```

Comment:
isn't the current behavior we'll take the best metric for all runs and only rely on the impute strategy if there is an issue grabbing the metric value.

I thought you said we will use the best metric only for `finished` runs and rely on the use specificed strategy for the other 3 cases.

Reply:
this was a relic of some experimentation, fixed with added testing